imap_data_access/io.py

```diff
@@ -173,7 +173,7 @@ def query(
         logger.debug("Decoded JSON: %s", items)
 
     # if latest version was included in search then filter returned query for largest.
-    if version == "latest":
+    if (version == "latest") and (len(items) > 0):
```

Comment:
As an FYI, you can also check against an empty list
```suggestion
    if (version == "latest") and items:
```